tslearn/clustering.py

```diff
@@ -981,9 +980,12 @@ def predict(self, X):
         """
         X = check_array(X, allow_nd=True, force_all_finite='allow-nan')
         check_is_fitted(self, 'cluster_centers_')
-        X = check_dims(X, self.cluster_centers_.shape)
-        X_ = to_time_series_dataset(X)
-        return self._assign(X_, update_class_attributes=False)
+        if self.metric == "euclidean":
```

Comment:
An idea would be to support variable lengths for euclidean distances as well, but this could be a future PR. It could be calculated similarly as shapelets by e.g. shifting the shorter timeseries over the longer one and taking the minimal distance.

Reply:
Hum, that would feel awkward to support variable length for Euclidean metric, I do not think I would be in favor of that.